service_layer/lti/OIDCLoginFlask.py

```diff
@@ -281,11 +281,14 @@ def lti_launch_from_id_token(self) -> Response:
         )
 
         try:
-            user = services.get_user_by_lms_id(
-                unit_of_work.SqlAlchemyUnitOfWork(), self.id_token.sub
-            )
-            logger.debug("User: " + str(user))
-            if not user:
+            try:
```

Comment:
lieber if else anstatt die exeception als if nutzen

Reply:
Das blöde ist, dass get_user_by_lms_id einen Fehler wirft wenn es keinen Findet. Sehe gerade keinen Weg zu prüfen ob es einen Nutzer schon gibt ohne das die Methode keinen Fehler schmeißt

Reply:
Deswegen hatte das vorherige "if not user" nie geklappt, da es nie dahin kam und sofort in die untere Exception gesprungen ist